### PR TITLE
[ioctl] Refactor aliasimport command in new ioctl

### LINKED ISSUE #3273: [ioctl] refactor aliasimport command in new ioctl
### What would you like to be added
 - Step 1: refactor  command  [newcmd/alias/aliasimport.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/alias/aliasimport.go) 
 - Step 2: append necessary unit test case in [newcmd/alias/aliasimport_test.go](https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/alias/aliasimport_test.go) 

#### Notes:
1. Use [function of client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) . 
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/alias/aliasimport.go ).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `aliasimport` command in new `ioctl`, with the following note.

* Update all function to [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go)
* Replace `fmt.Println` with `cmd.Println`
* [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. e.g. output.PrintError could be replaced with errors.Wrap.
* Global variables (config.ReadConfig config.DefaultConfigFile etc.) have been wrapped into [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go).
* Refactor unit test to cover the modification.

Fixes #3273 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] TestNewAliasImportCmd
- [x] TestNewAccountCreateAdd
- [x] TestNewAccountImportCmd
- [x] TestNewAccountImportKeyCmd
- [x] TestNewAccountImportKeyStoreCmd
- [x] TestNewAccountImportPemCmd
- [x] TestNewAliasSetCmd

**Test Configuration**:

- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/newcmd/alias/aliasimport_test.go [hunk 1]
```diff
@@ -12,68 +17,73 @@ import (
...
+		require.NotNil(result)
+	})
 
-	require.NoError(t, err)
-	require.NotNil(t, result)
+	t.Run("import alias with json format", func(t *testing.T) {
+		cmd := NewAliasImportCmd(client)
+		result, err := util.ExecuteCmd(cmd, `{"name":"test","address":"io1uwnr55vqmhf3xeg5phgurlyl702af6eju542sx"}`)
+		require.NoError(err)
+		require.NotNil(result)
```
Comment (huof6829) [inline ioctl/newcmd/alias/aliasimport_test.go:49]: can compare result with name or address in line 47 ?

File: ioctl/newcmd/alias/aliasimport_test.go [hunk 2]
```diff
@@ -12,68 +17,73 @@ import (
...
-
-	cmd := NewAliasImportCmd(client)
 
-	result, err := util.ExecuteCmd(cmd, `{"name":"test","address":"io1uwnr55vqmhf3xeg5phgurlyl702af6eju542sx"}`)
+	t.Run("invalid flag", func(t *testing.T) {
+		cmd := NewAliasImportCmd(client)
+		result, err := util.ExecuteCmd(cmd, "-f", "test", "test")
+		require.Error(err)
+		require.Contains(err.Error(), "invalid flag")
+		require.NotNil(result)
```
Comment (huof6829) [inline ioctl/newcmd/alias/aliasimport_test.go:42]: delete because meaningless when error occurs

File: ioctl/newcmd/alias/aliasimport_test.go [hunk 3]
```diff
@@ -12,68 +17,73 @@ import (
...
-	result, err = util.ExecuteCmd(cmd, "-f", "yaml", `aliases:
-name: mhs2
-  address: io19sdfxkwegeaenvxk2kjqf98al52gm56wa2eqks`)
-	require.Error(t, err)
-	require.NotNil(t, result)
+	t.Run("force import", func(t *testing.T) {
+		cmd := NewAliasImportCmd(client)
+		result, err := util.ExecuteCmd(cmd, "-F", `{"name":"io1uwnr55vqmhf3xeg5phgurlyl702af6eju542sx","address":"io1uwnr55vqmhf3xeg5phgurlyl702af6eju542sx"}`)
+		require.NoError(err)
+		require.NotNil(result)
```
Comment (huof6829) [inline ioctl/newcmd/alias/aliasimport_test.go:69]: can check result?

File: ioctl/newcmd/alias/aliasimport_test.go [hunk 4]
```diff
@@ -12,68 +17,73 @@ import (
...
-	result, err = util.ExecuteCmd(cmd, "-f", "test", "test")
-	require.Error(t, err)
-	require.NotNil(t, result)
-}
+	t.Run("failed to unmarshal json imported aliases", func(t *testing.T) {
+		cmd := NewAliasImportCmd(client)
+		result, err := util.ExecuteCmd(cmd, `""{"name":"d","address":""}""`)
+		require.Error(err)
+		require.Contains(err.Error(), "failed to unmarshal imported aliases")
+		require.NotNil(result)
```
Comment (huof6829) [inline ioctl/newcmd/alias/aliasimport_test.go:77]: delete

File: ioctl/newcmd/alias/aliasimport_test.go [hunk 5]
```diff
@@ -12,68 +17,73 @@ import (
...
-	str := "0/0 aliases imported\nExisted aliases: a c"
-	require.Equal(message.String(), str)
+	t.Run("failed to unmarshal yaml imported aliases", func(t *testing.T) {
+		cmd := NewAliasImportCmd(client)
+		result, err := util.ExecuteCmd(cmd, "-f", "yaml", `aliases:
+			name: mhs2
+			address: io19sdfxkwegeaenvxk2kjqf98al52gm56wa2eqks`)
+		require.Error(err)
+		require.Contains(err.Error(), "failed to unmarshal imported aliases")
+		require.NotNil(result)
```
Comment (huof6829) [inline ioctl/newcmd/alias/aliasimport_test.go:87]: delete

File: ioctl/newcmd/alias/aliasimport_test.go (hunk 3 above)
Comment (LuckyPigeon) [reply ioctl/newcmd/alias/aliasimport_test.go:68]: Seems using `force-import` will cause a syncronize issue, according to my trial, `force-import` argument parse will execute after the loop below is done.
https://github.com/iotexproject/iotex-core/blob/3f224285e5cec85162c3406f8597a390b8da5f78/ioctl/newcmd/alias/aliasimport.go#L76

Need a little help for fixing this sync issue, thx!

Comment (huof6829) [reply ioctl/newcmd/alias/aliasimport_test.go:68]: If you input command with -F,  the value of `force-import` is parsed to true, will run to line81; without -F, `force-import` is parsed to false by default, will run into line76. 
-F means to force update the value of the alias, which the input alias already exists in default config file (~/.config/ioctl/default/config.default).

**Review by huof6829 — APPROVED** [review]

File: ioctl/client.go [hunk 6]
```diff
@@ -242,22 +246,26 @@ func (c *client) AliasMap() map[string]string {
...
 	}
-	c.cfg.Aliases[alias] = addr
-	return c.writeAlias()
+	c.cfg.Aliases[aliasName] = addr
 }
 
-func (c *client) DeleteAlias(alias string) error {
-	delete(c.cfg.Aliases, alias)
-	return c.writeAlias()
+func (c *client) SetAlias(aliasName string, addr string) error {
```
Comment (Liuhaai) [inline ioctl/client.go:258]: `SetAliasAndSave`

File: ioctl/client.go [hunk 7]
```diff
@@ -242,22 +246,26 @@ func (c *client) AliasMap() map[string]string {
 	return aliases
 }
 
-func (c *client) SetAlias(alias string, addr string) error {
+func (c *client) SetAliasUnwritten(aliasName string, addr string) {
```
Comment (Liuhaai) [inline ioctl/client.go:249]: `SetAlias`

File: ioctl/newcmd/alias/aliasimport.go [hunk 8]
```diff
@@ -46,77 +49,67 @@ type importMessage struct {
...
+				if !forceImport && c.Config().Aliases[importedAlias.Name] != "" {
 					message.Unimported = append(message.Unimported, importedAlias)
 					continue
 				}
-				for aliases[importedAlias.Address] != "" {
-					delete(config.ReadConfig.Aliases, aliases[importedAlias.Address])
-					aliases = c.AliasMap()
-				}
-				config.ReadConfig.Aliases[importedAlias.Name] = importedAlias.Address
+				c.SetAlias(args[0], importedAlias.Address)
```
Comment (Liuhaai) [inline ioctl/newcmd/alias/aliasimport.go:92]: could `SetAliasAndSave` be used here?

Comment (LuckyPigeon) [reply ioctl/newcmd/alias/aliasimport.go:92]: So we didn't use `SetAliasAndSave` here, right?

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**